docs/aws/audit/esmonitoring/rules/es_whitelisted_ips_only.mdx

```diff
@@ -23,6 +23,255 @@ CBP
+   ```
+
+2. **Authorize Inbound Rules for the Security Group:**
+   Add inbound rules to the security group to allow access from the specific IP addresses.
+
+   ```sh
+   aws ec2 authorize-security-group-ingress --group-name my-elasticsearch-sg --protocol tcp --port 443 --cidr <WHITELISTED_IP>/32
+   ```
+
+   Repeat the above command for each whitelisted IP address.
+
+3. **Associate the Security Group with the Elasticsearch Domain:**
+   Modify the Elasticsearch domain to associate it with the newly created security group.
+
+   ```sh
+   aws es update-elasticsearch-domain-config --domain-name my-elasticsearch-domain --vpc-options SecurityGroupIds=<SECURITY_GROUP_ID>
+   ```
+
+4. **Verify the Configuration:**
+   Ensure that the Elasticsearch domain is now using the correct security group.
+
+   ```sh
+   aws es describe-elasticsearch-domain --domain-name my-elasticsearch-domain
+   ```
+
+   Check the `VPCOptions` section in the output to confirm that the correct security group is associated.
+
+By following these steps, you can ensure that your Elasticsearch domain is accessible only from the whitelisted IP addresses using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Elasticsearch domains from being accessible only from whitelisted IP addresses using Python scripts, you can use the AWS SDK for Python (Boto3). Below are the steps to achieve this:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with your AWS credentials and region.
+
+3. **Define the Access Policy**:
+   Create an access policy that allows access only from specific IP addresses.
+
+4. **Update the Elasticsearch Domain Access Policy**:
+   Use the `update_elasticsearch_domain_config` method to apply the access policy to your Elasticsearch domain.
+
+Here is a Python script that demonstrates these steps:
+
+```python
+import boto3
+import json
+
+# Step 1: Initialize a Boto3 session
+session = boto3.Session(
+    aws_access_key_id='YOUR_AWS_ACCESS_KEY_ID',
+    aws_secret_access_key='YOUR_AWS_SECRET_ACCESS_KEY',
+    region_name='YOUR_AWS_REGION'
+)
+
+# Step 2: Create an Elasticsearch client
+es_client = session.client('es')
+
+# Step 3: Define the access policy
+# Replace 'YOUR_ELASTICSEARCH_DOMAIN_NAME' with your domain name
+# Replace 'YOUR_WHITELISTED_IP' with the IP address you want to whitelist
+access_policy = {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": "*",
+            "Action": "es:*",
+            "Condition": {
+                "IpAddress": {
+                    "aws:SourceIp": "YOUR_WHITELISTED_IP"
+                }
+            },
+            "Resource": "arn:aws:es:YOUR_AWS_REGION:YOUR_AWS_ACCOUNT_ID:domain/YOUR_ELASTICSEARCH_DOMAIN_NAME/*"
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>